Build failed
```
2024-01-09 11:19:53 /ClickHouse/src/Common/TargetSpecific.cpp:43:13: error: enumeration values 'AMXBF16', 'AMXTILE', and 'AMXINT8' not handled in switch [-Werror,-Wswitch]
```

--- a/src/Common/CpuId.h
+++ b/src/Common/CpuId.h
@@ -94,7 +94,9 @@ inline bool cpuid(UInt32 op, UInt32 * res) noexcept /// NOLINT
     OP(CLWB)                 \
     OP(XSAVE)                \
     OP(OSXSAVE)              \
-    OP(AMX)
+    OP(AMXBF16)              \
+    OP(AMXTILE)              \
+    OP(AMXINT8)
 
 union CpuInfo
 {
@@ -182,17 +184,22 @@ bool haveOSXSAVE() noexcept
     return (CpuInfo(0x1).registers.ecx >> 27) & 1u;
 }
 
-bool haveAVX() noexcept
+inline bool haveSIMDExtention(UInt32 code) noexcept
 {
 #if defined(__x86_64__) || defined(__i386__)
     // http://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-optimization-manual.pdf
     // https://bugs.chromium.org/p/chromium/issues/detail?id=375968
     return haveOSXSAVE()                           // implies haveXSAVE()
            && (our_xgetbv(0) & 6u) == 6u              // XMM state and YMM state are enabled by OS
-           && ((CpuInfo(0x1).registers.ecx >> 28) & 1u); // AVX bit
+           && ((CpuInfo(0x1).registers.ecx >> code) & 1u); // Extention bit
 #else
     return false;
 #endif
+}
+
+bool haveAVX() noexcept
+{
+    return haveSIMDExtention(28); // AVX bit
 }
 
 bool haveFMA() noexcept
@@ -314,13 +321,19 @@ bool haveRDRAND() noexcept
     return CpuInfo(0x0).registers.eax >= 0x7 && ((CpuInfo(0x1).registers.ecx >> 30) & 1u);
 }
 
-bool haveAMX() noexcept
+bool haveAMXBF16() noexcept
 {
-#if defined(__x86_64__) || defined(__i386__)
-    return (CpuInfo(0x1).registers.ecx >> 22) & 1u;
-#else
-    return false;
-#endif
+    return haveSIMDExtention(22);  // AMX-BF16 bit
+}
+
+bool haveAMXTILE() noexcept
+{
+    return haveSIMDExtention(24);  // AMX-TILE bit
+}
+
+bool haveAMXINT8() noexcept
+{
+    return haveSIMDExtention(25);  // AMX-INT8 bit
 }
 
 struct CpuFlagsCache

--- a/src/Common/TargetSpecific.cpp
+++ b/src/Common/TargetSpecific.cpp
@@ -23,6 +23,12 @@ UInt32 getSupportedArchs()
         result |= static_cast<UInt32>(TargetArch::AVX512VBMI);
     if (Cpu::CpuFlagsCache::have_AVX512VBMI2)
         result |= static_cast<UInt32>(TargetArch::AVX512VBMI2);
+    if (Cpu::CpuFlagsCache::have_AMXBF16)
+        result |= static_cast<UInt32>(TargetArch::AMXBF16);
+    if (Cpu::CpuFlagsCache::have_AMXTILE)
+        result |= static_cast<UInt32>(TargetArch::AMXTILE);
+    if (Cpu::CpuFlagsCache::have_AMXINT8)
+        result |= static_cast<UInt32>(TargetArch::AMXINT8);
     return result;
 }
 

--- a/src/Common/TargetSpecific.cpp
+++ b/src/Common/TargetSpecific.cpp
@@ -50,6 +50,9 @@ String toString(TargetArch arch)
         case TargetArch::AVX512BW:    return "avx512bw";
         case TargetArch::AVX512VBMI:  return "avx512vbmi";
         case TargetArch::AVX512VBMI2: return "avx512vbmi2";
+        case TargetArch::AMXBF16: return "amxbf16";
+        case TargetArch::AMXTILE: return "amxtile";
+        case TargetArch::AMXINT8: return "amxint8";
     }
 
     UNREACHABLE();

--- a/src/Common/TargetSpecific.h
+++ b/src/Common/TargetSpecific.h
@@ -83,6 +83,9 @@ enum class TargetArch : UInt32
     AVX512BW    = (1 << 4),
     AVX512VBMI  = (1 << 5),
     AVX512VBMI2 = (1 << 6),
+    AMXBF16 = (1 << 7),
+    AMXTILE = (1 << 8),
+    AMXINT8 = (1 << 9),
 };
 
 /// Runtime detection.

